src/lxc/bdev/bdev.c

```diff
@@ -890,40 +890,41 @@ static struct bdev *bdev_get(const char *type)
 
 static const struct bdev_type *get_bdev_by_name(const char *name)
 {
-	size_t i;
+	size_t i, cmplen;
 
-	for (i = 0; i < numbdevs; i++) {
-		if (strcmp(bdevs[i].name, name) == 0)
-			return &bdevs[i];
-	}
+	cmplen = strcspn(name, ":");
+	if (cmplen == 0)
+		return NULL;
```

Comment:
if the `name` does not have prefix "xxxx:",  `strcspn` will return `strlen(name)`. so it would be better

```c
if (cmplen == 0 || cmplen == strlen(name))
               return NULL;
```